### ISSUE TITLE
Process killed during trimesh contains

### ISSUE THREAD

**Issue by sunwonlikeyou** [ZhanxyR/SHERT#8]
I did quick_demo.py with demo image.
But Process was killed because of calculating _contains_ method
in lib.tools.semantic_normal_sampling.py , line 346
Trimesh version is 3.23.5

```
inside = target_mesh.contains(src_verts).astype(int) + 1 
```

**Comment by ZhanxyR** [comment]
Hi, I'll check later. And what about the version of Open3D?

--------------原始邮件--------------
发件人："sunwonlikeyou ***@***.***&gt;;
发送时间：2024年10月23日(星期三) 下午5:52
收件人："ZhanxyR/SHERT" ***@***.***&gt;;
抄送："Subscribed ***@***.***&gt;;
主题：[ZhanxyR/SHERT] Process killed during trimesh contains (Issue #8)
-----------------------------------

 

 
I did quick_demo.py with demo image.
 But Process was killed because of calculating contains method
 in lib.tools.semantic_normal_sampling.py , line 346
 Trimesh version is 3.23.5
 inside = target_mesh.contains(src_verts).astype(int) + 1

**Comment by ZhanxyR** [comment]
I have checked that we also used the `trimesh==3.23.5`, could you please provide some more information about the crash.

**Comment by sunwonlikeyou** [comment]
open3d==0.18.0, torch==2.3.1+cu121.
0.10.0 version couldn't be installed.

**Comment by ZhanxyR** [comment]
I think the open3d can not be installed mainly due to the version of python. Are you using python&gt;3.8?

--------------原始邮件--------------
发件人："sunwonlikeyou ***@***.***&gt;;
发送时间：2024年10月24日(星期四) 上午8:24
收件人："ZhanxyR/SHERT" ***@***.***&gt;;
抄送："Xiaoyu Zhan ***@***.***&gt;;"Comment ***@***.***&gt;;
主题：Re: [ZhanxyR/SHERT] Process killed during trimesh contains (Issue #8)
-----------------------------------

 


open3d==0.18.0, torch==2.3.1+cu121.open3d==0.18.0,&nbsp;torch==2.3.1+cu121.
 0.10.0 version couldn't be installed.0.10.0版本无法安装。

—&amp;mdash;

**Comment by sunwonlikeyou** [comment]
No, I'm using python3.11. But I think there was no error, then code should be run.

**Comment by ZhanxyR** [comment]
There remain the bug when using `o3d.io.read_triangle_mesh` with a higher version of open3d mainly due to the invalid triangle indexes in our processed `.obj` files. We recommend to use `python==3.8` and `open3d==0.10`. If you hope to run the code with different settings, all of the called function should be fixed. You can try this function with different version of open3d in our subdivided mesh file to see the differences.

**Comment by sunwonlikeyou** [comment]
Thank you, I'll change the settings and then I tell you again.
